app/views/events/_form.html.erb

```diff
@@ -0,0 +1,50 @@
+<%= form_with(model: event) do |form| %>
+  <% if event.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(event.errors.count, "error") %> prohibited this event from being saved:</h2>
+
+      <ul>
+        <% event.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div class="field">
+    <%= form.label :name %>
+    <%= form.text_field :name %>
+  </div>
+
+  <div class="field">
+    <%= form.label :description %>
+    <%= form.text_area :description %>
+  </div>
+
+    <div>
+      <%= form.label :room %>
+        <%= form.select 'room_id',
+            options_from_collection_for_select(Room.all, :id, :name, event.room.nil? ? 1 : event.room.id) + options_for_select([["No Room", nil]]),
+            {},
+            multiple: false %>
+    </div>
+
+  <div class="field">
+    <%= form.label "Start Date" %>
+    <%= form.datetime_select :d_start %>
+  </div>
+
+  <div class="field">
+    <%= form.label "End Date" %>
+    <%= form.datetime_select :d_end %>
+  </div>
```

Comment:
There is a big time zone problem here, which can be fixed (at least if the server is run in the same time zone as the users) by inserting:
```ruby
Time.zone = Time.now.zone
```
Somewhere prominent in the code.
It is even possible to add it into the `config/application.rb` like this:
```ruby
config.time_zone = Time.now.zone
```